frontend/src/common-components/QrReader.tsx

```diff
@@ -8,27 +8,34 @@ interface QrReaderProps {
 export function QrReader({ onScan }: QrReaderProps) {
   const codeReader = useRef(new BrowserQRCodeReader())
   const videoElement = useRef<HTMLVideoElement>(null)
-  const controls = useRef<IScannerControls>()
-
-  async function decodeContinuously() {
-    if (!videoElement.current) return
-    const videoInputDevices = await BrowserCodeReader.listVideoInputDevices()
-    const selectedDeviceId = videoInputDevices[0].deviceId
-    controls.current = await codeReader.current.decodeFromVideoDevice(selectedDeviceId, videoElement.current, (result, err, controls) => {
-      console.debug(result, err)
-      if (result) {
-        controls.stop()
-        onScan(result.getText())
-      }
-    })
-  }
 
   useEffect(() => {
+    let controls: IScannerControls | null
+    let cancelled = false
+
+    async function decodeContinuously() {
+      if (!videoElement.current) return
+      const videoInputDevices = await BrowserCodeReader.listVideoInputDevices()
+      const selectedDeviceId = videoInputDevices[0].deviceId
+      if (cancelled) return
+      controls = await codeReader.current.decodeFromVideoDevice(selectedDeviceId, videoElement.current, (result, err, controls) => {
+        console.debug(result, err)
```

Comment:
You left a logging statement here

Reply:
Oh that was already there, maybe it's good to remove it

Reply:
It was already there